controller/dataplane/watch.go

```diff
@@ -79,3 +87,28 @@ func listDataPlanesReferencingKongPluginInstallation(
 		})
 	}
 }
+
+func listDataPlanesReferencingDataPlaneKonnectExtension(
+	c client.Client,
+) handler.TypedMapFunc[*operatorv1alpha1.DataPlaneKonnectExtension, reconcile.Request] {
+	return func(
+		ctx context.Context, ext *operatorv1alpha1.DataPlaneKonnectExtension,
+	) []reconcile.Request {
+		logger := ctrllog.FromContext(ctx)
+
+		// Find all DataPlane resources referencing DataPlaneKonnectExtension
+		// that maps to the DataPlaneKonnectExtension enqueued for reconciliation.
+		var dataPlaneList operatorv1beta1.DataPlaneList
+		if err := c.List(ctx, &dataPlaneList, client.MatchingFields{
+			index.DataPlaneKonnectExtensionIndex: ext.Namespace + "/" + ext.Name,
+		}); err != nil {
```

Comment:
Since we're disallowing cross namespace refs we can add the `client.InNamespace(ext.Namespace)` option here, right?

```suggestion
		if err := c.List(ctx, &dataPlaneList,
			client.InNamespace(ext.Namespace),
			client.MatchingFields{
				index.DataPlaneKonnectExtensionIndex: ext.Namespace + "/" + ext.Name,
			},
		); err != nil {
```

Reply:
It does not really matter, as the indexing is already done only for objects already living in the same namespace. Hence, all the objects returned by this list are already in the same ns.